src/ert/scheduler/scheduler.py

```diff
@@ -174,6 +191,32 @@ def add_dispatch_information_to_jobs_file(self) -> None:
         for job in self._jobs.values():
             self._update_jobs_json(job.iens, job.real.run_arg.runpath)
 
+    async def monitor_tasks(self, scheduling_tasks: list[asyncio.Task[None]]) -> None:
+        pending: Iterable[asyncio.Task[None]] = (
+            list(self._tasks.values()) + scheduling_tasks
+        )
+
+        while True:
+            done, pending = await asyncio.wait(
```

Comment:
is `pending` used, or can we change it to `_` if it is unused?

Reply:
Yes, it's used in `asyncio.wait`

Reply:
done, **pending** = await asyncio.wait(...)
Is this pending used?
We use done further down, but I don't see this pending being used anywhere.

Reply:
Can we use `while len(pending) > 0` instead of `while True`?

Reply:
Since `pending` contains also the scheduling_tasks, we can't use `pending == []`. There is `if not self.is_active():` which takes care of stopping this loop.